### PR TITLE
[Intermediate Representation] Translate analyzer IR to analysis

### LINKED ISSUE #975: [Intermediate Representation] Generate the analysis program
Will do this under the directory `sqlflow/pkg/sql/codegen/analysis/*`

### PR DESCRIPTION
**Pull request by weiguoz**
fix #975

### CODE REVIEW

File: pkg/sql/codegen/analyzer/codegen.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+func GenAnalysis(ir *codegen.AnalyzeIR, modelPath string) (string, error) {
+	if strings.HasPrefix(strings.ToUpper(ir.TrainIR.Estimator), "XGBOOST.") {
+		return genXGBAnalysis(ir, modelPath)
+	}
+	return "", fmt.Errorf("unsupported model:%s", ir.TrainIR.Estimator)
+}
+
+func genXGBAnalysis(ir *codegen.AnalyzeIR, modelPath string) (string, error) {
+	if ir.Explainer != "TreeExplainer" {
+		return "", fmt.Errorf("unsupported explainer")
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/analyzer/codegen.go]: => `fmt.Errorf("unsupported explainer %s", ir.Explainer)`

File: pkg/sql/codegen/analyzer/codegen.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+const (
+	shapSummaryAttributes = "shap_summary"
+)
+
+// GenAnalysis generates a Python program to analyze a trained model.
+func GenAnalysis(ir *codegen.AnalyzeIR, modelPath string) (string, error) {
+	if strings.HasPrefix(strings.ToUpper(ir.TrainIR.Estimator), "XGBOOST.") {
+		return genXGBAnalysis(ir, modelPath)
+	}
+	return "", fmt.Errorf("unsupported model:%s", ir.TrainIR.Estimator)
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/analyzer/codegen.go]: `model:%s` => `model: %s`

File: pkg/sql/codegen/analyzer/codegen.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"sqlflow.org/sqlflow/pkg/sql/codegen"
+)
+
+const (
+	shapSummaryAttributes = "shap_summary"
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/analyzer/codegen.go]: Please switch to `attribute.Dictionary` in the upcoming PRs. Example:

https://github.com/sql-machine-learning/sqlflow/blob/eb19db58b36640b58cca0e6b979be0dff7101e4a/pkg/sql/codegen/xgboost/codegen.go#L37-L47

File: pkg/sql/codegen/analyzer/codegen.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+)
+
+// GenAnalysis generates a Python program to analyze a trained model.
+func GenAnalysis(ir *codegen.AnalyzeIR, modelPath string) (string, error) {
```
Comment (tonyyang-svail) [inline pkg/sql/codegen/analyzer/codegen.go]: GenAnalysis => Analyze. This function should be called as `xgboost.Analyze`.

**Review by tonyyang-svail — CHANGES_REQUESTED** [review]
Please move `pkg/sql/codegen/analyzer/*.go` to `pkg/sql/codegen/xgboost/`.

It is important to keep the following mapping consistent.

1. TrainIR => xgboost.Train
1. PredictIR => xgboost.Predict
1. AnalyzeIR => xgboost.Analyze

And we can generate the documentation for train/pred/analyze XGBoost all together.

File: pkg/sql/codegen/xgboost/codegen_analyze.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		ModelFile:          modelPath,
+	}
+	var analysis bytes.Buffer
+	if err := analyzeTemplate.Execute(&analysis, fr); err != nil {
+		return "", err
+	}
+	return analysis.String(), nil
+}
+
+func resolveParames(attrs map[string]interface{}, group string) (map[string]interface{}, error) {
```
Comment (typhoonzero) [inline pkg/sql/codegen/xgboost/codegen_analyze.go]: resolveParames -> resolveParams

**Review by typhoonzero — COMMENTED** [review]
@tonyyang-svail @weiguoz  should we put this codegen under directory `codegen/shap` because it's generating SHAP code, just like `codegen/tensorflow`?

**Comment by weiguoz** [comment]
> @tonyyang-svail @weiguoz should we put this codegen under directory `codegen/shap` because it's generating SHAP code, just like `codegen/tensorflow`?

It looks intuitive to split directories by dependency packages. However, from the perspective of the model lifecycle, the current implementation is also reasonable. Plus, more code for the `PREDICT` process in codegen would be shared with the `ANALYZE` process, e.g. `getFieldMeta()` and `mockTrainIR()`.

**Comment by tonyyang-svail** [comment]
> @tonyyang-svail @weiguoz should we put this codegen under directory codegen/shap because it's generating SHAP code, just like codegen/tensorflow?

@typhoonzero We train/predict/analyze an XGBoost model, their documentation should be together, so the generation of analyzing code should in `xgboost` package.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**